### PR TITLE
Add SimpleCov.

### PR DESCRIPTION
**Pull request by squaresurf**
Why:

* SimpleCov will give us local insight into our test coverage and where
we can improve.

This change addresses the need by:

* Add SimpleCov gem to the test environment.
* Default the COVERAGE env var to on.
* Enable SimpleCov in the spec helper.

### CODE REVIEW

**Comment by Mrjaco12** [comment]
We should also add the cov/ dir to the .gitignore, I believe.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,3 +1,11 @@
+if ENV.fetch('COVERAGE', false)
+  require 'simplecov'
+  SimpleCov.start 'rails' do
+    add_filter '/app/admin/'
+    add_filter '/spec/'
```
Comment (samcdavid) [inline spec/spec_helper.rb]: I don't believe the spec folder has to be filtered out. I haven't had to filter it out for any of the projects where I have used simplecov.

**Comment by samcdavid** [comment]
+1 to @Mrjaco12 comment, the `coverage` directory needs to be added to the gitignore. Other than that it looks great.

**Comment by squaresurf** [comment]
@samcdavid I've fixed the spec helper to not exclude the `spec` folder. I had to do that on another project that used more than one test framework, but we shouldn't need that here as we only use rspec currently.

I've also added the coverage directory to the gitignore and cleaned it up a bit.

@samcdavid @Mrjaco12 would you please re-review?

**Comment by Mrjaco12** [comment]
:+1:

**Comment by samcdavid** [comment]
:+1:

**Pull request merged**